src/Symfony/Component/Security/Core/Signature/SignatureHasher.php

```diff
@@ -45,24 +45,48 @@ public function __construct(PropertyAccessorInterface $propertyAccessor, array $
     }
 
     /**
-     * Verifies the hash using the provided user and expire time.
+     * Verifies the hash using the provided user identifier and expire time.
+     *
+     * This method must be called before the user object is loaded from a provider.
      *
      * @param int    $expires The expiry time as a unix timestamp
      * @param string $hash    The plaintext hash provided by the request
      *
      * @throws InvalidSignatureException If the signature does not match the provided parameters
      * @throws ExpiredSignatureException If the signature is no longer valid
      */
-    public function verifySignatureHash(UserInterface $user, int $expires, string $hash): void
+    public function acceptSignatureHash(string $userIdentifier, int $expires, string $hash): void
```

Comment:
For your consideration, the name of this method can be a bit confusing. Maybe we can make it longer and call it `verifySignatureHashForUserIndentifier()`

Reply:
It would be also confusing to me, vs `verifySignatureHash`, which we cannot rename.